draft-ietf-quic-transport.md

```diff
@@ -2018,7 +2047,26 @@ implementations SHOULD treat the receipt of a frame that is not expressly
 permitted in the description of a state as a connection error
 ({{error-handling}}). Frames of unknown types are ignored.
 
-(TODO: QUIC_STREAM_NO_ERROR is a special case.  Write it up.)
+
+## Solicited State Transitions
+
+If an endpoint is no longer interested in the data being received, it MAY send a
+DISINTEREST frame on a stream in the "open" or "half-closed (local)" state to
+request closure of the stream in the opposite direction.  This typically
+indicates that the receiving application is no longer reading from the stream
+and all future data will be discarded upon receipt.
+
+STREAM frames received after sending DISINTEREST are still counted toward the
+connection and stream flow-control windows.  Even though these frames will be
+discarded, because they are sent before their sender receives the DISINTEREST,
+the sender will consider the frames to count against its flow-control windows.
+
+Upon receipt of a DISINTEREST frame, an endpoint SHOULD send a RST_STREAM with
+an error code of QUIC_RECEIVED_RST.  If the DISINTEREST frame is received on a
+stream that is already in the "half-closed (local)" or "closed" states, a
+RST_STREAM frame SHOULD still be sent and retransmission of previously-sent
+STREAM frames SHOULD be cancelled.
```

Comment:
It kind of _is_ squishy, at least in my mind.  DISINTEREST is a courtesy notification that your data is being ignored, so you might as well stop sending it.  Converting that to a MUST is to put more semantics on it than I had intended.  Notably, this is also hard behavior to verify in a peer -- you'd have to look for STREAM frames in a later packet than the ACK of the DISINTEREST, I suppose.